alg/sha1.c

```diff
@@ -391,6 +402,18 @@ HMAC_SHA1_Final(uint8_t digest[20], HMAC_SHA1_CTX * ctx)
 	insecure_memzero(ihash, 20);
```

Comment:
`HMAC_SHA1_Final` implicitly clears its context by calling `SHA1_Final` on `->ictx` and `->octx`.  It might be appropriate for it to call the new `HMAC_SHA1_Clear` as well, just for extra clarity.